keyboards/handwired/mingwahlu/mingnumpad/keymaps/default/keymap.c

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2020 mingwahlu
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT_numpad_6x4(
+        KC_AUDIO_MUTE, KC_AUDIO_VOL_DOWN, KC_AUDIO_VOL_UP, KC_PSCREEN,
+        KC_NUMLOCK, KC_KP_SLASH, KC_KP_ASTERISK, KC_KP_MINUS,
+        KC_KP_7, KC_KP_8, KC_KP_9,
+        KC_KP_4, KC_KP_5, KC_KP_6, KC_KP_PLUS,
+        KC_KP_1, KC_KP_2, KC_KP_3,
+        KC_KP_0, KC_KP_DOT, KC_KP_ENTER
+    ),
+    [_FN] = LAYOUT_numpad_6x4(
+        KC__MUTE, KC__VOLDOWN, KC__VOLUP, KC_PSCREEN,
+        KC_NUMLOCK, KC_KP_SLASH, KC_KP_ASTERISK, KC_KP_MINUS,
+        KC_KP_7, KC_KP_8, KC_KP_9,
+        KC_KP_6, KC_KP_5, KC_KP_4, KC_KP_PLUS,
+        KC_KP_3, KC_KP_2, KC_KP_1,
+        KC_KP_0, KC_KP_DOT, KC_KP_ENTER
+    )
```

Comment:
It looks better if you use the short form keycodes here, since it fits the standard 7 char length for all the keycodes. 

Additionally, there is no layer keycode to get to the FN layer.  
Also, the audio keycodes should work for all OS's. No need to be add the mac ones. 
```suggestion
    [_BASE] = LAYOUT_numpad_6x4(
        KC_MUTE, KC_VOLD, KC_VOLU, KC_PSCR,
        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
        KC_KP_7, KC_KP_8, KC_KP_9,
        KC_KP_4, KC_KP_5, KC_KP_6, KC_PMNS,
        KC_KP_1, KC_KP_2, KC_KP_3,
        KC_KP_0,          KC_PDOT, KC_PENT
    ),
```